src/libvirtApi/domain.js

```diff
@@ -506,66 +481,49 @@ export function domainDesktopConsole({
+    const diskXML = getDiskElemByTarget(domXml, target);
 
-                return call(connectionName, vmPath, 'org.libvirt.Domain', 'GetXMLDesc', [getXMLFlags], { timeout, type: 'u' });
-            })
-            .then(domInactiveXml => {
-                const diskInactiveXML = getDiskElemByTarget(domInactiveXml[0], target);
-                if (diskInactiveXML && persistent)
-                    detachFlags |= Enum.VIR_DOMAIN_AFFECT_CONFIG;
+    const [domInactiveXml] = await call(connectionName, vmPath, 'org.libvirt.Domain', 'GetXMLDesc', [Enum.VIR_DOMAIN_XML_INACTIVE], { timeout, type: 'u' });
 
-                return call(connectionName, vmPath, 'org.libvirt.Domain', 'DetachDevice', [diskXML, detachFlags], { timeout, type: 'su' });
-            });
+    const diskInactiveXML = getDiskElemByTarget(domInactiveXml, target);
+    if (diskInactiveXML && persistent)
+        detachFlags |= Enum.VIR_DOMAIN_AFFECT_CONFIG;
+
+    await call(connectionName, vmPath, 'org.libvirt.Domain', 'DetachDevice', [diskXML, detachFlags], { timeout, type: 'su' });
 }
 
 // Cannot use virt-xml until https://github.com/virt-manager/virt-manager/issues/357 is fixed
-export function domainDetachHostDevice({ connectionName, vmId, live, dev }) {
+export async function domainDetachHostDevice({ connectionName, vmId, live, dev }) {
     const source = getHostDevSourceObject(dev);
     if (!source)
-        return Promise.reject(new Error(`domainDetachHostDevice: could not determine device's source identifier`));
-
-    const hostDevPromises = [];
+        throw new Error("domainDetachHostDevice: could not determine device's source identifier");
 
     // hostdev's <address bus=... device=...> may be different between live XML and offline XML (or it may be present in live XML but missing in offline XML)
     // therefore we need to call DetachDevice twice with different hostdevXMLs, once for live XML and once for offline XML
-    hostDevPromises.push(
-        call(connectionName, vmId, 'org.libvirt.Domain', 'GetXMLDesc', [Enum.VIR_DOMAIN_XML_INACTIVE], { timeout, type: 'u' })
-                .then(domInactiveXml => {
-                    const hostdevInactiveXML = getHostDevElemBySource(domInactiveXml[0], source);
-                    if (hostdevInactiveXML)
-                        return call(connectionName, vmId, 'org.libvirt.Domain', 'DetachDevice', [hostdevInactiveXML, Enum.VIR_DOMAIN_AFFECT_CONFIG], { timeout, type: 'su' });
-                })
-    );
+    const [domInactiveXml] = await call(connectionName, vmId, 'org.libvirt.Domain', 'GetXMLDesc', [Enum.VIR_DOMAIN_XML_INACTIVE], { timeout, type: 'u' });
+    const hostdevInactiveXML = getHostDevElemBySource(domInactiveXml, source);
+    if (hostdevInactiveXML)
+        await call(connectionName, vmId, 'org.libvirt.Domain', 'DetachDevice', [hostdevInactiveXML, Enum.VIR_DOMAIN_AFFECT_CONFIG], { timeout, type: 'su' });
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1397-20240122-150846-61fe453f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)